docs/source/tutorial/glance.rst

```diff
@@ -0,0 +1,359 @@
+
+Code Breakdown
+~~~~~~~~~~~~
+
+Let's start our python program. Matplotlib is used for the graphs to show training progress.
+
+.. code-block:: python
+
+   #!/usr/bin/env python
+   
+   from __future__ import print_function
+   
+   try:
+       import matplotlib
+       matplotlib.use('Agg')
+   except ImportError:
+       pass
+
+Typical imports for a Chainer program. Links contain trainable parameters and functions do not.
+
+.. code-block:: python
+
+   import chainer
+   import chainer.functions as F
+   import chainer.links as L
+   from chainer import training, datasets
+   from chainer.training import extensions
+   
+   import numpy as np
+   import sklearn.preprocessing as sp
+   
+From the raw mushroom.csv, we format the data into a Chainer dataset. Chainer requires a numpy array for the features in the X matrix and a flattened array if the label is one-dimensional.
+
+.. code-block:: python
+
+   data_array = np.genfromtxt('mushrooms.csv', delimiter=',',dtype=str, skip_header=1)
+   labelEncoder = sp.LabelEncoder()
+   for col in range(data_array.shape[1]):
+       data_array[:, col] = labelEncoder.fit_transform(data_array[:, col])
+   
+   X = data_array[:, 1:].astype(np.float32)
+   Y = np.ndarray.flatten(data_array[:, 0].astype(np.int32))
+   train, test = datasets.split_dataset_random(datasets.TupleDataset(X, Y), 623)
+   
+Define the neural network. For our mushrooms, we'll use two fully-connected, hidden layers between the input and output layers.
+
+.. code-block:: python
+
+   # Network definition
+   class MLP(chainer.Chain):
+   
+       def __init__(self, n_units, n_out):
+   	super(MLP, self).__init__()
+   	with self.init_scope():
+   	    # the size of the inputs to each layer will be inferred
+   	    self.l1 = L.Linear(n_units)  # n_in -> n_units
+   	    self.l2 = L.Linear(n_units)  # n_units -> n_units
+   	    self.l3 = L.Linear(n_out)  # n_units -> n_out
+   
+As an activation function, we'll use standard Rectified Linear Units (relu).
+
+.. code-block:: python
+
+       def __call__(self, x):
+   	h1 = F.relu(self.l1(x))
+   	h2 = F.relu(self.l2(h1))
+   	return self.l3(h2)
+   
+Since mushrooms are either edible or poisonous (no information on psychedelic effects!) in the dataset, we'll use a classifier Link for the output, with 100 units in the hidden layers and 2 possible categories to be classified into.
+   
+.. code-block:: python
+
+   model = L.Classifier(MLP(100, 2))
+
+If using a CPU instead of the GPU, set gpu_id to -1. Otherwise, use the ID of the GPU, usually 0.
+
+.. code-block:: python
+
+   gpu_id = -1
+   
+   if gpu_id >= 0:
+       # Make a specified GPU current
+       chainer.cuda.get_device_from_id(gpu_id).use()
+       model.to_gpu()  # Copy the model to the GPU
+   
+Pick and optimizer, and set up the model to use it.
+
+.. code-block:: python
+
+   # Setup an optimizer
+   optimizer = chainer.optimizers.SGD()
```

Comment:
In the whole code block above, `Adam` is used. Please unify what optimizer you use.